lib/UmdMainTemplatePlugin.js

```diff
@@ -124,16 +134,16 @@ class UmdMainTemplatePlugin {
 				) +
 				"	else if(typeof define === 'function' && define.amd)\n" +
 				(requiredExternals.length > 0 ?
-					(this.name && this.namedDefine === true ?
-						"		define(" + libraryName(this.name) + ", " + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n" :
+					(this.names.amd && this.namedDefine === true ?
+						"		define(" + libraryName(this.names.amd) + ", " + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n" :
 						"		define(" + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n"
 					) :
-					(this.name && this.namedDefine === true ?
-						"		define(" + libraryName(this.name) + ", [], " + amdFactory + ");\n" :
+					(this.names.amd && this.namedDefine === true ?
+						"		define(" + libraryName(this.names.amd) + ", [], " + amdFactory + ");\n" :
```

Comment:
@sokra Not sure how to get that line covered. Looks like it wasn't before either, and this PR now improves coverage overall. https://codecov.io/gh/webpack/webpack/pull/4883/changes